connman/gweb/gweb.c

```diff
@@ -475,6 +477,30 @@ static inline void call_result_func(struct web_session *session, guint16 status)
 
 }
 
+static inline int call_dns_route_func(struct web_session *session, bool add)
+{
+	bool ret;
+
+	if (add == session->dns_routes_set)
+		return -EALREADY;
+
+	if (session->dns_route_func) {
```

Comment:
if route func is null, function returns "succeeded" without doing anything / even toggling the state -> a bit fishy.

Reply:
Yeah, it should change state. Will change that one.

Reply:
Done in [a4e3c22](https://github.com/sailfishos/connman/pull/55/commits/a4e3c2202bab7ec34a107eac78bab9c12b6567ff)